token-lending/program/src/processor.rs

```diff
@@ -2010,6 +2015,29 @@ fn process_update_reserve_config(
         msg!("Lending market owner provided must be a signer");
         return Err(LendingError::InvalidSigner.into());
     }
+    // if it's a permissionless market
+    if &solend_market_owner::id() != lending_market_owner_info.key {
+        if reserve.config.protocol_liquidation_fee != config.protocol_liquidation_fee {
+            msg!("permissionless markets can't edit protocol liquidation fees");
+            return Err(LendingError::InvalidConfig.into());
+        }
+        if reserve.config.protocol_take_rate != config.protocol_take_rate {
+            msg!("permissionless markets can't edit protocol take rate");
+            return Err(LendingError::InvalidConfig.into());
+        }
+        if reserve.config.fee_receiver != config.fee_receiver {
+            msg!("permissionless markets can't edit fee receiver");
+            return Err(LendingError::InvalidConfig.into());
+        }
+        if reserve.config.fees.flash_loan_fee_wad != config.fees.flash_loan_fee_wad {
+            msg!("permissionless markets can't edit flash loan fees");
+            return Err(LendingError::InvalidConfig.into());
+        }
+        if reserve.config.fees.host_fee_percentage != config.fees.host_fee_percentage {
+            msg!("permissionless markets can't edit host fee percentage");
+            return Err(LendingError::InvalidConfig.into());
+        }
```

Comment:
is it intentional that origination fees can be changed by permissionless pool owner?

Reply:
yeh

Reply:
tests are failing bc i think we try to set one of these config params with a non-5ph key. apart from that, lgtm

Reply:
yeh i was gonna fix that soon